Add hyphenation_decompounder token filter docs #8062 (#8120)
opensearch-project/documentation-website: _analyzers/token-filters/hyphenation-decompounder.md

---
layout: default
title: Hyphenation decompounder
parent: Token filters
nav_order: 170
---

> # Hyphenation decompounder token filter

Reviewer: ```suggestion

# Hyphenation decompounder token filter

```

---
layout: default
title: Hyphenation decompounder
parent: Token filters
nav_order: 170
---

> <!-- vale off -->

Reviewer: ```suggestion

```

# Hyphenation decompounder token filter

> <!-- vale on -->

Reviewer: ```suggestion

```

> The `hyphenation_decompounder` token filter is used to break down compound words into their constituent parts. This filter is particularly useful for languages like German, Dutch, and Swedish, where compound words are common. The filter uses hyphenation patterns (typically defined in .xml files) to identify the possible locations within a compound word where it can be split into components. These components are then checked against a provided dictionary. If there is a match, those components are treated as valid tokens.

Reviewer: ```suggestion
The `hyphenation_decompounder` token filter is used to break down compound words into their constituent parts. This filter is particularly useful for languages like German, Dutch, and Swedish, where compound words are common. The filter uses hyphenation patterns (typically defined in .xml files) to identify the possible locations within a compound word where it can be split into components. These components are then checked against a provided dictionary. If there is a match, those components are treated as valid tokens. For more information about hyphenation pattern files, see [FOP XML Hyphenation Patterns](https://offo.sourceforge.net/#FOP+XML+Hyphenation+Patterns).
```

> The `hyphenation_decompounder` token filter is used to break down compound words into their constituent parts. This filter is particularly useful for languages like German, Dutch, and Swedish, where compound words are common. The filter uses hyphenation patterns (typically defined in .xml files) to identify the possible locations within a compound word where it can be split into components. These components are then checked against a provided dictionary. If there is a match, those components are treated as valid tokens.

Reviewer: Can we provide a link to an example xml file that contains hyphenation pattern rules?

> The `hyphenation_decompounder` token filter is used to break down compound words into their constituent parts. This filter is particularly useful for languages like German, Dutch, and Swedish, where compound words are common. The filter uses hyphenation patterns (typically defined in .xml files) to identify the possible locations within a compound word where it can be split into components. These components are then checked against a provided dictionary. If there is a match, those components are treated as valid tokens. For more information about hyphenation pattern files, see [FOP XML Hyphenation Patterns](https://offo.sourceforge.net/#FOP+XML+Hyphenation+Patterns).

Reviewer 2: ```suggestion
The `hyphenation_decompounder` token filter is used to break down compound words into their constituent parts. This filter is particularly useful for languages like German, Dutch, and Swedish, in which compound words are common. The filter uses hyphenation patterns (typically defined in .xml files) to identify the possible locations within a compound word where it can be split into components. These components are then checked against a provided dictionary. If there is a match, those components are treated as valid tokens. For more information about hyphenation pattern files, see [FOP XML Hyphenation Patterns](https://offo.sourceforge.net/#FOP+XML+Hyphenation+Patterns).
```

## Parameters

The `hyphenation_decompounder` token filter can be configured with the following parameters.

Parameter | Required/Optional | Data type | Description
:--- | :--- | :--- | :--- 

> `hyphenation_patterns_path` | Required | String | Path (relative to `config` directory or absolute) to the hyphenation patterns file, which contains the language-specific rules for where to split words. The file is typically in XML format. Sample files can be downloaded from [OFFO Sourceforge project](https://sourceforge.net/projects/offo/).

Reviewer: ```suggestion
`hyphenation_patterns_path` | Required | String | The path (relative to the `config` directory or absolute) to the hyphenation patterns file, which contains the language-specific rules for word splitting. The file is typically in XML format. Sample files can be downloaded from [OFFO Sourceforge project](https://sourceforge.net/projects/offo/).
```

> `hyphenation_patterns_path` | Required | String | The path (relative to the `config` directory or absolute) to the hyphenation patterns file, which contains the language-specific rules for word splitting. The file is typically in XML format. Sample files can be downloaded from [OFFO Sourceforge project](https://sourceforge.net/projects/offo/).

Reviewer 2: ```suggestion
`hyphenation_patterns_path` | Required | String | The path (relative to the `config` directory or absolute) to the hyphenation patterns file, which contains the language-specific rules for word splitting. The file is typically in XML format. Sample files can be downloaded from the [OFFO SourceForge project](https://sourceforge.net/projects/offo/).
```

> `word_list` | Required if `word_list_path` is not set | Array of strings | A list of words that are used to validate the components generated by the hyphenation patterns.

Reviewer 2: ```suggestion
`word_list` | Required if `word_list_path` is not set | Array of strings | A list of words used to validate the components generated by the hyphenation patterns.
```

> `word_list_path` | Required if `word_list` is not set | String | Path (relative to `config` directory or absolute) to list of subwords.

Reviewer: ```suggestion
`word_list_path` | Required if `word_list` is not set | String | The path (relative to `config` directory or absolute) to a list of subwords.
```

> `word_list_path` | Required if `word_list` is not set | String | The path (relative to `config` directory or absolute) to a list of subwords.

Reviewer 2: ```suggestion
`word_list_path` | Required if `word_list` is not set | String | The path (relative to the `config` directory or absolute) to a list of subwords.
```

> `max_subword_size` | Optional | Integer | Maximum subword length. If the generated subword is longer, it will not be added to the generated tokens. Default is `15`.

Reviewer 2: ```suggestion
`max_subword_size` | Optional | Integer | The maximum subword length. If the generated subword exceeds this length, it will not be added to the generated tokens. Default is `15`.
```

> `min_subword_size` | Optional | Integer | Minimum subword length. If the generated subword is shorted, it will not be added to the generated tokens. Default is `2`.

Reviewer: ```suggestion
`min_subword_size` | Optional | Integer | Minimum subword length. If the generated subword is shorter, it will not be added to the generated tokens. Default is `2`.
```

> `min_subword_size` | Optional | Integer | Minimum subword length. If the generated subword is shorter, it will not be added to the generated tokens. Default is `2`.

Reviewer 2: ```suggestion
`min_subword_size` | Optional | Integer | The minimum subword length. If the generated subword is shorter than the specified length, it will not be added to the generated tokens. Default is `2`.
```

`min_word_size` | Optional | Integer | Minimum word character length. Shorter word tokens are excluded from decomposition into subwords. Default is `5`.

> `only_longest_match` | Optional | Boolean | Only include the longest subword in the generated tokens. Default os `false`.

Reviewer: ```suggestion
`only_longest_match` | Optional | Boolean | Only include the longest subword in the generated tokens. Default is `false`.
```

> `min_word_size` | Optional | Integer | Minimum word character length. Shorter word tokens are excluded from decomposition into subwords. Default is `5`.

Reviewer 2: ```suggestion
`min_word_size` | Optional | Integer | The minimum word character length. Word tokens shorter than this length are excluded from decomposition into subwords. Default is `5`.
```

> `only_longest_match` | Optional | Boolean | Only include the longest subword in the generated tokens. Default is `false`.

Reviewer 2: ```suggestion
`only_longest_match` | Optional | Boolean | Only includes the longest subword in the generated tokens. Default is `false`.
```

## Example

> The following example request creates a new index named `test_index` and configures an analyzer with `hyphenation_decompounder` filter:

Reviewer: ```suggestion
The following example request creates a new index named `test_index` and configures an analyzer with a `hyphenation_decompounder` filter:
```
